### ISSUE TITLE
False Positive With Null-Coalesce Operator

### ISSUE THREAD

**Issue by iakoder** [kalessil/phpinspectionsea#1438]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 3.0.16.2 |
| Language level | e.g. PHP 7.0                        |

Current behaviour
--
```
<?php

/**
 * @return array
 */
function retArray() {
    return [];
}

/**
 * @param array      $a
 * @param array|null $x
 */
function t(array &$a, array $x = null) {

    // No issues
    $a = $x ?? retArray();

    // New value type (null) is not in annotated types
    $a = ( $x ?? retArray() );
}
```
Expected behaviour
--
Adding () around the ?? operator shouldn't affect type info.  I think the version without the () was an issue in older versions of the plugin?  (I'm seeing reports from our CI environment that uses an older version).  I ran across the issue with () when trying to narrow down why inspections were being reported there.

Environment details
---
PhpStorm 2019.2.3
Build #PS-192.6817.20, built on September 25, 2019
Licensed to Intacct / Dan Wilks
Subscription is active until June 23, 2020
Runtime version: 11.0.4+10-b304.69 x86_64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
macOS 10.14.6
GC: ParNew, ConcurrentMarkSweep
Memory: 4029M
Cores: 8
Registry: run.processes.with.pty=TRUE, ide.mnemonic.helper.old=true, debugger.watches.in.variables=false
Non-Bundled Plugins: com.jetbrains.upsource, org.jetbrains.plugins.stylus, com.kalessil.phpStorm.phpInspectionsEA

**Comment by kalessil** [comment]
That was a very nice finding - a bug in out types resolver, thanks! Fixed.
